crates/dbsp/src/operator/dynamic/balance/maxsat.rs

```diff
@@ -0,0 +1,451 @@
+//! MaxSat solver.
```

Comment:
This should be in a library, with a well-defined API so it can be replaced with a more capable solver if necessary.